main.sh

```diff
@@ -33,6 +35,7 @@ post () {
   fi
 
   # POST to federalist's build finished endpoint && POST to federalist-builder's build finished endpoint
+  set +o pipefail
```

Comment:
Do you want to `set -o pipefail` before the clone.sh/build.sh/publish.sh trio below?

You mention "When /app/build.sh fails there, we want the entire build to fail." but pipefail will be disabled from here until the end of the script.